### PR TITLE
Add first iteration of tests to repo

### PR DESCRIPTION
**Pull request by robbibt**
This adds some very rudimentary tests to the repo!
image

Currently they use a small pickle file of satellite data to test against the `elevation` function, and downloads a zip of tide modelling data. In the future we can improve it by:

- Replacing the pickle with prod database access
- Syncing tide models directly from S3

Ignore the long list of commits - we can merge this with **squash and merge**. The ECR sync step will fail on this branch, but hopefully work on `main`.

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -14,7 +14,11 @@ pyTMD==2.0.5
 pandas==1.5.3
 pygeos==0.14
 pyproj==3.4.1
+pytest
+pytest-dependency
+pytest-cov
 pytz==2023.3
+rioxarray
```
Comment (robbibt) [inline requirements.in:21]: Had to add back `rioxarray` - `xarray` used it under the hood after all

File: tests/data/satellite_ds.pickle
Comment (robbibt) [inline tests/data/satellite_ds.pickle:1]: Pickle file of satellite data - to replace with database access later on

File: intertidal/elevation.py [hunk 2]
```diff
@@ -1494,44 +1495,55 @@ def intertidal_cli(
...
-            time_range=(start_date, end_date),
-            resolution=resolution,
-            crs="EPSG:3577",
-            include_s2=True,
-            include_ls=True,
-            filter_gqa=True,
-            max_cloudcover=90,
-            skip_broken_datasets=True,
-        )
+        if study_area == "testing":
```
Comment (robbibt) [inline intertidal/elevation.py:1507]: Use pickle file instead of loading data from datacube if study area == "testing"

File: codecov.yaml
Comment (robbibt) [inline codecov.yaml:1]: Ignore the validation and composites modules in code coverage for now

File: .github/workflows/dea-intertidal-image.yml [hunk 3]
```diff
@@ -30,13 +33,33 @@ jobs:
...
+        with:
+          fetch-depth: 0
 
       - name: Build DEA Intertidal image
         timeout-minutes: 20
         shell: bash
         run: |
           docker-compose build
 
+      - name: Run tests
```
Comment (robbibt) [inline .github/workflows/dea-intertidal-image.yml:45]: This downloads tide data from Dropbox (to replace with S3 access to tide modelling files), then runs some tests using `docker compose run`. Then finally upload code coverage to `codecov`

File: .github/workflows/dea-intertidal-image.yml [hunk 4]
```diff
@@ -30,13 +33,33 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```
Comment (robbibt) [inline .github/workflows/dea-intertidal-image.yml:37]: Maybe makes the git close faster?

File: .github/workflows/dea-intertidal-image.yml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-name: DEA Intertidal Image Push
+name: Image build and test
```
Comment (robbibt) [inline .github/workflows/dea-intertidal-image.yml:1]: Renamed so it shows up nicely on the readme badges

File: intertidal/extents.py [hunk 6]
```diff
@@ -77,7 +77,7 @@ def extents(
     freq,
     dem,
     corr,
-    land_use_mask="/gdata1/data/land_use/ABARES_CLUM/geotiff_clum_50m1220m/clum_50m1220m.tif",
+    land_use_mask="https://dea-public-data-dev.s3-ap-southeast-2.amazonaws.com/abares_clum_2020/clum_50m1220m.tiff",
```
Comment (robbibt) [inline intertidal/extents.py:80]: Load data from S3 so it works in the tests

File: intertidal/exposure.py [hunk 7]
```diff
@@ -14,8 +14,6 @@
 # from pyproj import Transformer
 # from scipy.signal import argrelmax, argrelmin
 
-from scipy.interpolate import interp1d
```
Comment (robbibt) [inline intertidal/exposure.py:17]: Not used below; removing for now

**Review by vnewey — APPROVED** [review]
Looks great Robbi!

**Pull request merged**